src/PIL/ImageGrab.py

```diff
@@ -149,29 +149,27 @@ def grabclipboard():
             session_type = None
 
         if shutil.which("wl-paste") and session_type in ("wayland", None):
-            output = subprocess.check_output(["wl-paste", "-l"]).decode()
-            mimetypes = output.splitlines()
-            if "image/png" in mimetypes:
-                mimetype = "image/png"
-            elif mimetypes:
-                mimetype = mimetypes[0]
-            else:
-                mimetype = None
-
-            args = ["wl-paste"]
-            if mimetype:
-                args.extend(["-t", mimetype])
+            args = ["wl-paste", "-t", "image"]
         elif shutil.which("xclip") and session_type in ("x11", None):
             args = ["xclip", "-selection", "clipboard", "-t", "image/png", "-o"]
         else:
             msg = "wl-paste or xclip is required for ImageGrab.grabclipboard() on Linux"
             raise NotImplementedError(msg)
 
         p = subprocess.run(args, capture_output=True)
-        err = p.stderr
-        if err:
-            msg = f"{args[0]} error: {err.strip().decode()}"
+        err = p.stderr.decode()
+        if p.returncode != 0:
+            allowed_errors = [
+                "Nothing is copied",  # wl-paste, when the clipboard is empty
+                "not available",  # wl-paste, when an image isn't available
+                "cannot convert",  # xclip, when an image isn't available
+                "There is no owner",  # xclip, when the clipboard isn't initialized
```

Comment:
Could you describe how I can manually reproduce this error?

Reply:
I can reproduce that by logging oit of gnome, and logging back in. Then run the xclip command. I can confirm this happens on fedora 39.